File: Shashank/1365.cpp
class Solution {
public:
    vector<int> smallerNumbersThanCurrent(vector<int>& nums) {
        int cnt[101] = {0};
        for(int i = 0;i < nums.size();i++){
            cnt[nums[i]]++;
        }
        for(int i = 1; i < 101;i++){
            cnt[i] += cnt[i-1];
        }
        for(int i = 0; i < nums.size();i++){            
            nums[i] = (nums[i])?cnt[nums[i]-1]:0;
        }
        return nums;
    }
};




    
    
    
    

File: Pratyasha/1365.py
from typing import List

class Solution:
    def smallerNumbersThanCurrent(self, nums: List[int]) -> List[int]:
        counts = [0] * 101
        for num in nums:
            counts[num] += 1
            print("num=",num,"counts=",counts[num])
        print(counts)
        counts.insert(0, 0)
        print(counts)
        for i in range(1,len(counts)):
            print("before:", i,counts[i])
            counts[i] = counts[i-1] + counts[i]
            print("after",i,counts[i])
            
        return [counts[num] for num in nums]
       
Solution().smallerNumbersThanCurrent([8,1,2,2,3]) 
